[server] Prevent deletion of default database to avoid catalog open failure

### Search before asking

- [x] I searched in the [issues](https://github.com/apache/fluss/issues) and found nothing similar.


### Description

### Summary  
Currently, MetadataManager.dropDatabase() does not check whether the database being dropped is the default database (fluss). This allows users to accidentally delete the default database, which causes Flink Catalog to fail to open with:

> Caused by: org.apache.flink.table.catalog.exceptions.CatalogException: The configured default-database 'fluss' does not exist in the Fluss cluster.
	at org.apache.fluss.flink.catalog.FlinkCatalog.open(FlinkCatalog.java:151)

  Furthermore, restarting CoordinatorServer will not recreate the default database, because createDefaultDatabase() only creates it when
  databases.isEmpty() returns true — if other databases exist, the default database is never recreated.

###   Steps to Reproduce

  1. Create a Fluss cluster (default database fluss is auto-created)
  2. Create another database (e.g., my_db)
  3. Drop the default database: DROP DATABASE fluss
  4. Try to open Flink Catalog with 'default-database' = 'fluss' → fails
  5. Restart CoordinatorServer → default database is not recreated (because my_db still exists)

###   Expected Behavior

 dropDatabase() should reject attempts to delete the default database with a clear error message.

 
### Proposed Fix

  Add a check in MetadataManager.dropDatabase() before proceeding with deletion:

```  
public void dropDatabase(String name, boolean ignoreIfNotExists, boolean cascade)
          throws DatabaseNotExistException, DatabaseNotEmptyException {
      if (CoordinatorServer.DEFAULT_DATABASE.equals(name)) {
          throw new UnsupportedOperationException(
                  "Cannot drop the default database '" + name
                          + "'. The default database is required for cluster operation.");
      }
      // ... existing logic
  }
```

### Willingness to contribute

- [x] I'm willing to submit a PR!

--- a/fluss-flink/fluss-flink-common/src/test/java/org/apache/fluss/flink/catalog/FlinkCatalogTest.java
+++ b/fluss-flink/fluss-flink-common/src/test/java/org/apache/fluss/flink/catalog/FlinkCatalogTest.java
@@ -196,19 +196,23 @@ class FlinkCatalogTest {
                         mockLakeCatalog);
         catalog.open();
 
-        // First check if database exists, and drop it if it does
+        // Clean up any leftover tables from previous tests
         if (catalog.databaseExists(DEFAULT_DB)) {
-            catalog.dropDatabase(DEFAULT_DB, true, true);
-        }
-        try {
-            catalog.createDatabase(
-                    DEFAULT_DB, new CatalogDatabaseImpl(Collections.emptyMap(), null), true);
-        } catch (CatalogException e) {
-            // the auto partitioned manager may create the db zk node
-            // in another thread, so if exception is NodeExistsException, just ignore
-            if (!ExceptionUtils.findThrowableWithMessage(e, "KeeperException$NodeExistsException")
-                    .isPresent()) {
-                throw e;
+            for (String table : catalog.listTables(DEFAULT_DB)) {
+                catalog.dropTable(new ObjectPath(DEFAULT_DB, table), true);
+            }
+        } else {
+            try {
+                catalog.createDatabase(
+                        DEFAULT_DB, new CatalogDatabaseImpl(Collections.emptyMap(), null), true);
+            } catch (CatalogException e) {
+                // the auto partitioned manager may create the db zk node
+                // in another thread, so if exception is NodeExistsException, just ignore
+                if (!ExceptionUtils.findThrowableWithMessage(
+                                e, "KeeperException$NodeExistsException")
+                        .isPresent()) {
+                    throw e;
+                }
             }
         }
     }

--- a/fluss-server/src/main/java/org/apache/fluss/server/coordinator/MetadataManager.java
+++ b/fluss-server/src/main/java/org/apache/fluss/server/coordinator/MetadataManager.java
@@ -314,6 +314,12 @@ public class MetadataManager {
 
     public void dropDatabase(String name, boolean ignoreIfNotExists, boolean cascade)
             throws DatabaseNotExistException, DatabaseNotEmptyException {
+        if (CoordinatorServer.DEFAULT_DATABASE.equals(name)) {
+            throw new UnsupportedOperationException(
+                    "Cannot drop the default database '"
+                            + name
+                            + "'. The default database is required for cluster operation.");
+        }
         if (!databaseExists(name)) {
             if (ignoreIfNotExists) {
                 return;

